c7n/mu.py

```diff
@@ -199,6 +199,20 @@ def close(self):
                 1024.0 * 1024.0)))
         return self
 
+    @staticmethod
+    def _temporary_opener(name, flag, mode=0o777):
+        return os.open(name, flag | os.O_TEMPORARY, mode)
+
+    def open(self):
+        """Return a file stream based on current platform"""
```

Comment:
I think for some platforms it is important to ensure temp file is closed before opening.

Reply:
To clarify - you are saying that we should make sure the package is closed ```self._closed = True``` 
before we attempt to open it? 

Similar to what is done here:
https://github.com/cloud-custodian/cloud-custodian/blob/a997d3a903239912c73f1c1dd338e89d6d02b3b6/c7n/mu.py#L213-L216

Reply:
You are absolutely correct. I did not realize the path property we are pulling from is derived from the same _temp_archive_file. I'll update accordingly. Good catch!